core/templates/pages/exploration-player-page/learner-experience/ratings-and-recommendations.component.spec.ts

```diff
@@ -107,12 +137,34 @@ describe('Ratings and recommendations component', () => {
     mockOnRatingUpdated.emit();
     tick();
 
+    expect(explorationPlayerStateService.isInStoryChapterMode)
+      .toHaveBeenCalled();
+    expect(componentInstance.inStoryMode).toBe(true);
+    expect(urlInterpolationService.interpolateUrl).toHaveBeenCalled();
+    expect(componentInstance.storyViewerUrl).toBe(
+      'dummy_story_viewer_page_url');
     expect(componentInstance.userRating).toEqual(userRating);
     expect(alertsService.addSuccessMessage).toHaveBeenCalled();
     expect(learnerViewRatingService.getUserRating).toHaveBeenCalled();
     expect(componentInstance.collectionId).toEqual(collectionId);
   }));
 
+  it('should not generate story page url if not in story mode',
+    fakeAsync(() => {
+      spyOn(explorationPlayerStateService, 'isInStoryChapterMode')
+        .and.returnValue(false);
+      spyOn(urlInterpolationService, 'interpolateUrl').and.returnValue(
+        'dummy_story_viewer_page_url');
+
+      componentInstance.ngOnInit();
+      tick();
+
+      expect(explorationPlayerStateService.isInStoryChapterMode)
+        .toHaveBeenCalled();
+      expect(componentInstance.inStoryMode).toBe(false);
```

Comment:
Do we need a precheck for this?

Reply:
Added a precheck for inStoryMode and storyViewerUrl